lib/omniauth/strategies/shopify.rb

```diff
@@ -23,6 +23,31 @@ def valid_site?
         !!(/\A(https|http)\:\/\/[a-zA-Z0-9][a-zA-Z0-9\-]*\.#{Regexp.quote(options[:myshopify_domain])}[\/]?\z/ =~ options[:client_options][:site])
       end
 
+      def valid_signature?
+        return false unless request.POST.empty?
+
+        params = request.GET
+        signature = params['hmac']
+        timestamp = params['timestamp']
+        return false unless signature && timestamp
+
+        return false unless timestamp.to_i > Time.now.to_i - 5 * 60
```

Comment:
In that case I think we should fix the clock synching, or pull the node out of rotation, since it seems like that could cause a lot of other problems. I don't know of any actual vulnerability involving replay attacks that is stopped by this timestamp. We already prevent the code being re-used in the access token endpoint.